api/src/main/java/org/apache/druid/indexer/TaskStatusPlus.java

```diff
@@ -49,6 +51,7 @@ public TaskStatusPlus(
       @JsonProperty("createdTime") DateTime createdTime,
       @JsonProperty("queueInsertionTime") DateTime queueInsertionTime,
       @JsonProperty("statusCode") @Nullable TaskState state,
+      @Deprecated @JsonProperty("status") @Nullable TaskState status,  // present for backwards compatibility
```

Comment:
For deserialization safety, can you add something here like:
- if "status" is null and "statusCode" is not null, set both to the value of "statusCode"
- If "statusCode" is null and "status" is not null, set both to the value of "status"
- if both are non-null, but have different values, log an error

I don't think there should be a situation where new code would be trying to deserialize an old TaskStatusPlus if the rolling upgrade sequence is followed, but I think it'd be good to handle that situation

Reply:
added the checks